### DISCUSSION TITLE
longer paths in metapath2vec

### DISCUSSION THREAD

**Discussion by nabihach** [pyg-team/pytorch_geometric#4052] [Q&A] [answered]
I'm trying to use PyG's MetaPath2Vec and I have a question about this line:

https://github.com/pyg-team/pytorch_geometric/blob/58eb6b433b4100544a20c05b5db4b8168a25d6c9/torch_geometric/nn/models/metapath2vec.py#L30

Why must each metapath be a 3-tuple? How can I specify longer metapaths, e.g. a metapath which has 5 items node_A --> Link A_A -> node_B --> link B_C --> node_C?

**Comment by rusty1s** [comment, marked as answer]
Yes, that is well possible. Note that `metapath` is a list of tuples, e.g.:
```python
[
    (author, writes, paper),
    (paper, cites, paper),
    (paper, written_by, author),
    ...
]
```

**Comment by nabihach** [comment]
I see, so the tuples together specify one metapath.
My follow-up question would be, how do I specify more than 1 metapaths?

**Reply by rusty1s** [reply]
That is not possible. Does the original metapath2vec model provide this functionality?

**Reply by nabihach** [reply]
Reading the paper again, I now realize it only uses a single metapath at a time. Thanks for answering and clearing it up!
